core/model/src/main/java/org/eclipse/rdf4j/model/vocabulary/XSD.java

```diff
@@ -0,0 +1,187 @@
+
+	/** <tt>http://www.w3.org/2001/XMLSchema#dayTimeDuration</tt> */
+	public static final IRI DAYTIMEDURATION = create("dayTimeDuration");
+
+	/** <tt>http://www.w3.org/2001/XMLSchema#time</tt> */
+	public final static IRI TIME = create("time");
+
+	/** <tt>http://www.w3.org/2001/XMLSchema#date</tt> */
+	public final static IRI DATE = create("date");
+
+	/** <tt>http://www.w3.org/2001/XMLSchema#gYearMonth</tt> */
+	public final static IRI GYEARMONTH = create("gYearMonth");
+
+	/** <tt>http://www.w3.org/2001/XMLSchema#gYear</tt> */
+	public final static IRI GYEAR = create("gYear");
+
+	/** <tt>http://www.w3.org/2001/XMLSchema#gMonthDay</tt> */
+	public final static IRI GMONTHDAY = create("gMonthDay");
+
+	/** <tt>http://www.w3.org/2001/XMLSchema#gDay</tt> */
+	public final static IRI GDAY = create("gDay");
+
+	/** <tt>http://www.w3.org/2001/XMLSchema#gMonth</tt> */
+	public final static IRI GMONTH = create("gMonth");
+
+	/** <tt>http://www.w3.org/2001/XMLSchema#string</tt> */
+	public final static IRI STRING = create("string");
+
+	/** <tt>http://www.w3.org/2001/XMLSchema#boolean</tt> */
+	public final static IRI BOOLEAN = create("boolean");
+
+	/** <tt>http://www.w3.org/2001/XMLSchema#base64Binary</tt> */
+	public final static IRI BASE64BINARY = create("base64Binary");
+
+	/** <tt>http://www.w3.org/2001/XMLSchema#hexBinary</tt> */
+	public final static IRI HEXBINARY = create("hexBinary");
+
+	/** <tt>http://www.w3.org/2001/XMLSchema#float</tt> */
+	public final static IRI FLOAT = create("float");
+
+	/** <tt>http://www.w3.org/2001/XMLSchema#decimal</tt> */
+	public final static IRI DECIMAL = create("decimal");
+
+	/** <tt>http://www.w3.org/2001/XMLSchema#double</tt> */
+	public final static IRI DOUBLE = create("double");
+
+	/** <tt>http://www.w3.org/2001/XMLSchema#anyURI</tt> */
+	public final static IRI ANYURI = create("anyURI");
+
+	/** <tt>http://www.w3.org/2001/XMLSchema#QName</tt> */
+	public final static IRI QNAME = create("QName");
+
+	/** <tt>http://www.w3.org/2001/XMLSchema#NOTATION</tt> */
+	public final static IRI NOTATION = create("NOTATION");
+
+	/*
+	 * Derived datatypes
+	 */
+
+	/** <tt>http://www.w3.org/2001/XMLSchema#normalizedString</tt> */
+	public final static IRI NORMALIZEDSTRING = create("normalizedString");
+
+	/** <tt>http://www.w3.org/2001/XMLSchema#token</tt> */
+	public final static IRI TOKEN = create("token");
+
+	/** <tt>http://www.w3.org/2001/XMLSchema#language</tt> */
+	public final static IRI LANGUAGE = create("language");
+
+	/** <tt>http://www.w3.org/2001/XMLSchema#NMTOKEN</tt> */
+	public final static IRI NMTOKEN = create("NMTOKEN");
+
+	/** <tt>http://www.w3.org/2001/XMLSchema#NMTOKENS</tt> */
+	public final static IRI NMTOKENS = create("NMTOKENS");
+
+	/** <tt>http://www.w3.org/2001/XMLSchema#Name</tt> */
+	public final static IRI NAME = create("Name");
+
+	/** <tt>http://www.w3.org/2001/XMLSchema#NCName</tt> */
+	public final static IRI NCNAME = create("NCName");
```

Comment:
I also prefer this structure over the one where we have a separate static block to instantiate the field.

Btw. I still think the use of uppercase for our fields is absolutely horrible. It is convention over common sense. This is particularly bad for the SHACL namespace, eg. sh:nodeKind and sh:NodeKind.

Reply:
I don't think we envisaged different case having different meanings when we originally adopted the convention for this. You're right though. I wouldn't mind changing it but we'll have to do them all in one go, and of course keep old all-uppercase variants around for backward compatibility.